app/controllers/accept_account_shares_controller.rb

```diff
@@ -1,12 +1,14 @@
 # frozen_string_literal: true
 
 class AcceptAccountSharesController < ApplicationController
+  before_action :received_share, only: [:show], if: -> { user_signed_in? }
+
   def show
     return if user_signed_in?
 
     session[:after_sign_in_url] = request.fullpath
 
-    shared_email = AccountShare.find_by!(token: ps.fetch(:token)).email
+    shared_email = AccountShare.unaccepted.find_by!(token: ps.fetch(:token)).email
```

Comment:
It's better, but seems it doesn't handle the case when I accept the created link by myself. It should also show 404 in this case. Also please add tests